src/main/java/ManageInputs.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    public static void dealWithDeadline(ArrayList<Task> tasks, int index, String line) throws UnexpectedCommandException, IOException {
+        int indexBy = line.indexOf("by");
+        int space = line.indexOf(" ");
+        if (indexBy == -1) {//invalid format
+            System.out.println("Invalid format! Enter deadline in the format: deadline (description) by (deadline)");
+            throw new UnexpectedCommandException();
+        }
+        try {//deadline / both not specified
+            String by = line.substring(indexBy + 2);
+        } catch (IndexOutOfBoundsException e) {
+            try {
+                String description = line.substring(space, indexBy - 1);
+            } catch (IndexOutOfBoundsException f) {
+                System.out.println("deadline description and deadline not specified");
+                throw new UnexpectedCommandException();
+            }
+            System.out.println("deadline not specified");
+            throw new UnexpectedCommandException();
+        }
+        try {//deadline not specified
+            String description = line.substring(space, indexBy - 1);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("deadline description not specified");
+            throw new UnexpectedCommandException();
+        }
+        String description = line.substring(space, indexBy - 1);
+        String by = line.substring(indexBy + 2);
+
+        tasks.add(index, new Deadline(description, by));
+    }
+
+    public static void dealWithTodo(ArrayList<Task> tasks, int index, String line) throws UnexpectedCommandException, IOException {
+        int indexSpace = line.indexOf(" ");
+        if (indexSpace == -1) {
+            System.out.println("todo description not specified");
+            throw new UnexpectedCommandException();
+        }
+
+        String description = line.substring(indexSpace);
+
+        tasks.add(index, new Todo(description));
+
+    }
+
+    private void handleUnexpectedCommand(boolean isValidCommand) throws UnexpectedCommandException {
+        if (!isValidCommand) {
+            throw new UnexpectedCommandException();
+        }
+    }
+
+    private void handleEmptyInput(String line) throws EmptyLineException {
+        if (line.isEmpty()) {
+            throw new EmptyLineException();
+        }
+    }
+
+    public ManageInputs(ArrayList<Task> tasks, int index, String line) throws IOException, UnexpectedCommandException {
+        index = fillFileContents(tasks, "TaskList.txt", index);
+        while (!line.equals("bye")) {
+            Boolean isValidCommand = false;
+            Scanner input = new Scanner(System.in);
+            line = input.nextLine();
+
+            Task t = new Task(line);
+            String[] inputs = line.split(" ");
+            
+            if (inputs[0].equals("mark")) {//mark as done
+                isValidCommand = true;
+                int idx = Integer.parseInt(inputs[1]);
```

Comment:
`index` and `idx` are being used in the code but seem to be referring to different things. It's not too clear what they really mean! Do consider  how you could make this function more readable!